static/src/javascripts/projects/common/modules/ui/cookiesBanner.js

```diff
@@ -10,18 +11,27 @@ import { Message } from 'common/modules/ui/message';
  * Show only on FIRST page view
  * Persist close state
  */
-const init = (): boolean => {
-    const geoContinentCookie = getCookie('GU_geo_continent');
-    if (!geoContinentCookie || geoContinentCookie.toUpperCase() !== 'EU') {
-        return false;
-    }
-
-    const EU_COOKIE_MSG = 'GU_EU_MSG';
-    const euMessageCookie = getCookie(EU_COOKIE_MSG);
-    if (euMessageCookie && euMessageCookie === 'seen') {
-        return false;
-    }
 
+const EU_COOKIE_MSG = 'GU_EU_MSG';
+
+const canShow = (): Promise<boolean> =>
+    new Promise(resolve => {
+        const geoContinentCookie = getCookie('GU_geo_continent');
+
+        if (!geoContinentCookie || geoContinentCookie.toUpperCase() !== 'EU') {
+            resolve(false);
+        }
+
+        const euMessageCookie = getCookie(EU_COOKIE_MSG);
+
+        if (euMessageCookie && euMessageCookie === 'seen') {
+            resolve(false);
+        } else {
+            resolve(true);
```

Comment:
Assuming there's nothing async happening in this callback, we can probably dispense with the wrapper `Promise` and return `Promise.resolve(true)` or `Promise.resolve(false)` as appropriate. What do you think?

Reply:
Yep, good suggestion 👍